utils/src/main/scala/com/salesforce/op/utils/spark/OpSparkListener.scala

```diff
@@ -91,14 +94,19 @@ class OpSparkListener
     appStartTime = appStartTime,
     appEndTime = appEndTime,
     appDuration = appEndTime - appStartTime,
-    stageMetrics = stageMetrics.toList,
+    stageMetrics = if (collectStageMetrics) stageMetrics.toList else List(cumulativeStageMetrics),
     versionInfo = VersionInfo()
   )
 
   override def onStageCompleted(stageCompleted: SparkListenerStageCompleted): Unit = {
     val si = stageCompleted.stageInfo
     val tm = si.taskMetrics
-    if (collectStageMetrics) stageMetrics += StageMetrics(si)
+    if (collectStageMetrics) {
+      stageMetrics += StageMetrics(si)
+    } else {
+      implicit val stageSG = caseclass.semigroup[StageMetrics]
+      cumulativeStageMetrics = cumulativeStageMetrics + StageMetrics(si)
```

Comment:
since you go this route you need to make some of the fields a dedicated Semigroup and override `plus` for it:
https://github.com/salesforce/TransmogrifAI/pull/341/files/3c25e7cf6581df8b29832b211acfb671b399d4f7#diff-862ee01af3e20d3ea939a0f9b89e362bR229

Reply:
see all fields that aren't `+`-ed

Reply:
tests would help to reveal the truth ;)